src/WebServerManager.cpp

```
@@ -930,108 +935,87 @@ void WebServerManager::handleApiWriteTag() {
+        fields.has_initial_weight = true;
+        fields.initial_weight_g = initial_weight_g;
     }
 
-    // 5. Consumed weight (only if both remaining_g and initial_weight_g provided)
     if (doc.containsKey("remaining_g") && doc.containsKey("initial_weight_g")) {
-        req.request_id = base_id + 4;
-        req.type = NFCWriteType::SET_CONSUMED_WEIGHT;
-        req.data.consumed_weight = consumed_g;
-        NFCManager::getInstance().enqueueWrite(req);
+        float consumed_g = initial_weight_g - remaining_g;
+        if (consumed_g < 0.0f) consumed_g = 0.0f;
```

Clang (14.0.6)

[warning] floating point literal has suffix 'f', which is not uppercase

(readability-uppercase-literal-suffix)